src/core/server/graph/resolvers/UserStatus.ts

```diff
@@ -53,10 +81,14 @@ export const UserStatus: Required<GQLUserStatusTypeResolver<
     ...user.consolidateUsernameStatus(username),
     userID,
   }),
-  ban: ({ ban, userID }): BanStatusInput => ({
-    ...user.consolidateUserBanStatus(ban),
-    userID,
-  }),
+  ban: async ({ ban, userID }, args, ctx): Promise<BanStatusInput> => {
+    const banStatus = user.consolidateUserBanStatus(ban, ctx.now, ctx.site?.id);
+    return {
+      ...banStatus,
+      sites: await loadSites(ctx, banStatus.siteIDs),
```

Comment:
This could instead be written as:

```suggestion
      sites: await ctx.loaders.Sites.site.loadMany(banStatus.siteIDs),
```

Because the `Site` resolver will automatically handle adding the `canModerate` and `topStories` for you.